src/transformation/builtins/array.ts

```diff
@@ -29,6 +30,8 @@ export function transformArrayConstructorCall(
     }
 }
 
+const lua50TableLength = lua.createTableIndexExpression(lua.createIdentifier("table"), lua.createStringLiteral("getn"));
```

Comment:
you should probably only make this when it is requested, and add the last parameter (tsOriginal node)

Reply:
Changed to a function with https://github.com/TypeScriptToLua/TypeScriptToLua/pull/1263/commits/d1bf66a73f8ec284d93cd6de239847d699c2c566. I don't believe `node` needs to be passed to the first usage, as it's passed to the assignment expression instead.